sklearn/ensemble/_weight_boosting.py

```diff
@@ -137,8 +137,15 @@ def fit(self, X, y, sample_weight=None):
         # Initializion of the random number instance that will be used to
         # generate a seed at each iteration
         random_state = check_random_state(self.random_state)
+        epsilon = np.finfo(np.float64).eps
```

Comment:
This should be dependent on the data type of `sample_weight`:

```suggestion
        epsilon = np.finfo(sample_weight.dtype).eps
```

but maybe we don't really support float32 sample weights. Anyway I think the code would be less arbitrary this way.

Reply:
I agree the code looks better this way.

And under the current implementation we have explicitly set `sample_weight` to be float64 with:
`sample_weight = _check_sample_weight(sample_weight, X, np.float64)`